src/core/net/srp_server.cpp

```diff
@@ -424,7 +424,7 @@ void Server::Start(void)
     VerifyOrExit(!IsRunning());
 
     SuccessOrExit(error = mSocket.Open(HandleUdpReceive, this));
-    SuccessOrExit(error = mSocket.Bind(0));
+    SuccessOrExit(error = mSocket.Bind(0, OT_NETIF_UNSPECIFIED));
```

Comment:
I think the SRP server should bind to Thread interface.

It doesn't make much sense for the SRP server to accept registrations
from both Thread network and Wi-Fi/Ethernet networks. At least, there
is currently no such use case I think.

Besides, I think the SRP server (and the DNSSD server) should not bind a
platform UDP socket at all because they don't work on both networks. The
only case that I see platform UDP is explicitly required is the Border Agent
that connects external Commissioner and the Leader (and Joiner). For other
general UDP services, I think the expected default behavior is that they should
only work in Thread network unless explicitly required (we may have a flag in
`Udp::Socket` that indicates whether we want to bind to a platform UDP). I think
it is safer and simpler in this way (e.g. we don't need to explicitly filter out TMF
and MLE ports) ,  Thoughts?
@bukepo @jwhui @simonlingoogle @librasungirl

Reply:
This PR focuses on dealing with the multicast issue reported from openthread-users. This change just keeps SRP server's current behavior. I don't have strong opinion on whether the SRP server should be accessible from non-Thread interfaces, but I remember Thread end device may choose any SRP server on the hub network in early discussions. Anyway, that's beyond the purpose of this PR. Should deal with that problem in separate PRs(such as https://github.com/openthread/openthread/pull/6285)?

Reply:
Sure, just came up with some thoughts around this.

Reply:
Let's just use this place for discussion. 

> Besides, I think the SRP server (and the DNSSD server) should not bind a platform UDP socket at all because they don't work on both networks.

Agree that most sockets don't even need to bind to the Thread TUN interface.

> (we may have a flag in Udp::Socket that indicates whether we want to bind to a platform UDP). I think
it is safer and simpler in this way (e.g. we don't need to explicitly filter out TMF and MLE ports). 

Agree that the currently method of listing all UDP ports that do not use PLAT_UDP in `ShouldUsePlatformUdp` is a little bit hardcoding. One solution I thought of is to check Thread UDP sockets first and only forward to Host if the UDP packet is not processed by Thread UDP sockets. But I think this would require significant (and difficult) refactoring to `Ip6::HandleDatagram`.

Reply:
I did some work with the similar idea in https://github.com/openthread/openthread/compare/main...bukepo:refactor/message/read-only-flag